lib/cErrors.ml

```diff
@@ -128,9 +128,14 @@ let print_gen ~anomaly (e, info) =
     str "<in exception printer>:" ++ spc() ++ print_anomaly anomaly exn ++ print_extra info ++ fnl() ++
     str "<original exception>:" ++ spc() ++ print_anomaly false e ++ extra_msg
 
+let print_quickfix = function
+  | [] -> mt ()
+  | qf -> fnl () ++ Quickfix.print qf
+
 (** The standard exception printer *)
 let iprint (e, info) =
-  print_gen ~anomaly:true (e,info)
+  print_gen ~anomaly:true (e,info) ++
+  Result.fold ~ok:print_quickfix ~error:(print_gen ~anomaly:true) (Quickfix.from_exception e)
```

Comment:
>UserError is already used elsewhere, eg Ltac, while the code above should really throw a dedicated exception (with an explicit Pp, a loc, a trace and Quickfix). I'd rather leave this cleanup to another PR.

OK but we should at least check test_mode here then
(I guess the test_mode ref would need to move down to lib)

Reply:
Frankly, I don't know what it is was not in Flags in the first place. Can I put it there?

Reply:
I think we generally try to avoid putting stuff in flags (it's seen by the kernel) but for this it should be OK

Reply:
still need to check test_mode here

Reply:
done